### PR TITLE
EdShel/features/implementRefreshTokens

### PR DESCRIPTION
**Pull request by EdShel**
Implemented refresh tokens sent by cookies.
Made docummentation for api/users in readme.md

### CODE REVIEW

File: UniAtHome/UniAtHome.DAL/Repositories/UsersRepository.cs [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
-        private UserManager<User> userManager;
+        private readonly UniAtHomeDbContext context;
 
-        public UsersRepository(UserManager<User> userManager)
+        private readonly UserManager<User> userManager;
+
+        private readonly int refreshTokenLifetime;
+
+        public UsersRepository(
+            UniAtHomeDbContext context,
```
Comment (lesloress) [inline UniAtHome/UniAtHome.DAL/Repositories/UsersRepository.cs]: Don't use UniAtHomeDbContext. We should use DbContext and then make Set<T>. (IoC)

File: UniAtHome/UniAtHome.BLL/DTOs/Auth/LoginResponse.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.AspNetCore.Identity;
+using System.Collections.Generic;
+
+namespace UniAtHome.BLL.DTOs.Auth
+{
+    public sealed class LoginResponse : ErrorProneOperationResponse
```
Comment (arstrybuh257) [inline UniAtHome/UniAtHome.BLL/DTOs/Auth/LoginResponse.cs:6]: Please, put responses entities into Web.API project

**Review by lesloress — CHANGES_REQUESTED** [review]

**Review by arstrybuh257 — CHANGES_REQUESTED** [review]

File: UniAtHome/UniAtHome.BLL/DTOs/Auth/LoginResponse.cs (hunk 2 above)
Comment (EdShel) [reply UniAtHome/UniAtHome.BLL/DTOs/Auth/LoginResponse.cs:6]: Why? Responses are used only as the results of services' requests. They ain't designed to be used the other way. In addition BLL should not depend on the WEB Api project (it's even impossible to make)

**Comment by EdShel** [comment]
There is a brand new branch with Blackjack and refresh tokens

**Pull request closed without merging**